include/eve/traits/common_value.hpp

```diff
@@ -13,15 +13,42 @@
 
 namespace eve::detail
 {
+  template<typename T>
+  struct fth { using type = T; };
+
+  template<scalar_value S0, scalar_value S1>
+  consteval fth<decltype((std::declval<S0>() + std::declval<S1>()))> operator%(fth<S0>, fth<S1>) noexcept
+  {
+    return {};
+  }
+
+  template<typename T, typename N, scalar_value S>
+  consteval auto operator%(fth<wide<T, N>>, fth<S>) noexcept
+  {
+    return fth<wide<T, N>>{};
+  }
+
+  template<typename T, typename N, scalar_value S>
+  consteval auto operator%(fth<S>, fth<wide<T, N>>) noexcept
+  {
+    return fth<wide<T, N>>{};
+  }
+
+  template<typename T, typename N>
+  consteval auto operator%(fth<wide<T, N>>, fth<wide<T, N>>) noexcept
+  {
+    return fth<wide<T, N>>{};
+  }
+
   template<typename... Ts>
-  using find_type = decltype(( std::declval<Ts>() + ... ));
+  using find_type = typename decltype((fth<Ts>{} % ...))::type;
```

Comment:
1. The name `fth` doesn't mean anyhthing. And it get's exposed everywhere.
2. My find_type has the same problem, we should name it better.  
3. don't pollute the namespace, operators should be done through a `hidden friends idiom`
4. Don't do the overloading, just do if constexpr chain.

```
template <typename T> 
struct find_common_value_combinator {
  using type = T;
  
 template <typename U> 
  friend auto operator%(
    find_common_value_combinator x,
    find_common_value_combinator<U> y) {
    if constexpr (...) {
   }
  }
};
```

Reply:
I don't like it. Because we don't have local names in headers - it's better to have the word `common` in there somewhere.

Reply:
find_common_type? common_value_find? I'm not sure why we called a type value, but that's what it is already.